lec11-modelling.Rmd

```diff
@@ -0,0 +1,527 @@
+
+Now, let's fit a model to the data. Let's say we want to see the role that sex
+has on weight in various species in the `portal` dataset. (`gaussian` is stating 
+that `weight` is a continuous variable and that you assume the error terms have
+a gaussian, or normal, distribution.)
+
+```{r}
+fit1 <- glm(weight ~ sex, data = portal, family = gaussian)
+summary(fit1)
+```
+
+There's a lot of results and information contained within `fit1`, which
+`summary()` extracts and presents in a fairly nice format. However, a lot of
+this information we aren't really interested in. So let's extract only what we
+want, using the `broom` package. (get it, broom to clean up?)
+
+```{r}
+tidy(fit1)
+```
+
+Much nicer! Ok, so what does this actually mean? Let's go back to the equation
+and convert the above into the formula:
+
+$$ weight = Intercept + (sexM \times b) $$
+
+Let's substitute in the numbers. The intercept, is well, the intercept in the
+data above. The $b$ is the estimate for `sex`.
+
+$$ weight = 42.17 + (sexM \times 0.825) $$
+
+We can use this equation to estimate the value of weight for a specific sex. But
+first, let's make sure this is working correctly. Let's find out the mean weight
+by sex.
+
+```{r}
+portal %>% 
+    filter(!is.na(sex)) %>% 
+    group_by(sex) %>% 
+    summarize(mean_wt = mean(weight, na.rm = TRUE)) 
+```
+
+Ok, let's compare to the equation. The variable `sexM` is 1 for male and 0 for
+female. So if we want to calculate the weight of female, we have to set `sexM`
+as 0 in the equation. Let's do it:
+
+```{r}
+42.17 + (0 * 0.825)
+```
+
+Same as above! And for males (equal to 1):
+
+```{r}
+42.17 + (1 * 0.825)
+```
+
+Good, it matches what we calculated. Why is this useful? Well, we can add more
+terms to the GLM equation. This allows us to determine how a variable influences
+the $y$ when other variables are held constant and is known as *multiple linear
+regression*. Since weight is probably also dependent on `hindfoot_length`, let's
+add that to the model and put the values into the equation.
+
+```{r}
+fit2 <- glm(weight ~ sex + hindfoot_length, data = portal)
+tidy(fit2)
+```
+
+$$ weight = -32.09 + (sexM \times -1.72) + (hindfoot\_length \times 2.56)$$
+
+Hmm, that's interesting... is a *negative* intercept... How can weight be
+negative? Well, that's because the intercept is calculated by setting *both* sex
+and hindfoot length to zero! But is it physically even possible that a hindfoot be
+non-existant?? No. But math doesn't care what is physically possible or not.
+That's where your ecology/biology knowledge comes in. So let's change that up by
+changing the actual value of hindfoot length in the dataset. There are several
+ways to transform the data into something that is meaningful to us. But first:
+
+> What does everyone think is a meaningful way to transform hindfoot length so
+we can interpret the model more appropriately?
+
+There are many ways to do it, but one way that is often used is to mean center
```

Comment:
center -> centre 

Canadian spelling woot woot!

Reply:
in this case, mean centering is a common statistical term. I tend to think of centre as a place or building, and in stats, center is to the data.

Reply:
@mbonsma Where were you for the colour/color debate?

Reply:
Yeah, yeah, it's one thing to use `color` in the code, quite another to spell it the wrong way in the text. lol

Reply:
Luckily, R supports both, generally...